### PR TITLE
feat(forge): add cheatcode vm.difficulty()

### PR DESCRIPTION
**Pull request by sinasab**
## Motivation

Wanting to mock `block.difficulty`.

## Solution

Adding a new cheatcode `vm.difficulty(uint256 newDifficulty)` to mock `block.difficulty`.

### CODE REVIEW

**Comment by sinasab** [comment]
Draft to add it to the docs repo @ https://github.com/foundry-rs/book/pull/536 for after this is settled

File: testdata/cheats/Cheats.sol [hunk 1]
```diff
@@ -6,6 +6,8 @@ interface Cheats {
     struct Log {bytes32[] topics; bytes data;}
     // Set block.timestamp (newTimestamp)
     function warp(uint256) external;
+    // Set block.difficulty (newDifficulty)
+    function whelm(uint256) external;
```
Comment (mds1) [inline testdata/cheats/Cheats.sol:10]: I'd like to suggest a more intuitive name 😅
```suggestion
    function difficulty(uint256) external;
```

**Review by mattsse — CHANGES_REQUESTED** [review]
impl and test lgtm, ty

but would also suggest to call this function simply `difficulty`, that would be self-explanatory

File: testdata/cheats/Cheats.sol (hunk 1 above)
Comment (sinasab) [reply testdata/cheats/Cheats.sol]: Changed

**Review by mattsse — APPROVED** [review]

**Comment by mds1** [comment]
Thanks @sinasab! Do you mind also submitting a PR to forge-std which adds the cheatcode to `Vm.sol`: https://github.com/foundry-rs/forge-std/blob/master/src/Vm.sol

**Pull request merged**